src/main/java/org/tron/core/net/node/Item.java

```diff
@@ -21,7 +21,7 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (!(o instanceof Item)) {
+    if (this.getClass() != o.getClass()) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "o" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=853&issues=AWOg829fmXhJxGlVU5lN&open=AWOg829fmXhJxGlVU5lN)

[ISSUE_KEY]: AWOg829fmXhJxGlVU5lN